### PR TITLE
Add Libadwaita widgets and improve existing widgets

### PR DESCRIPTION
**Pull request by david-swift**
Hello @JCWasmx86,

Thanks for starting this project! I added some new Libadwaita widgets, extended existing widgets with properties and functions, and made some general changes.

Here is a demo, showcasing some of the newly supported widgets and functions:

https://github.com/JCWasmx86/SwiftGui/assets/106754840/55425366-1883-43d4-ab81-62814e882668

<details>
<summary><b>List of the changes</b></summary>

- New functions and properties for `Window`, `App`, `NativeWidgetPeer`
- Computed properties for converting Swift types (`Bool`, `String`, `Int`, `Array`) to C types
- `gtui_stringlist_append` has the return type `void` instead of `uint64_t`
- Fix `gtui_box_prepend` (use `gtk_box_prepend` instead of `gtk_box_append`)
- Update both demos
- Default values for the `hexpand` and `vexpand` functions (`true`)
- Organize the content of `GTUI` with folders
- Add `StatusPage`
- Add `ToastOverlay` and `Toast`
- Add `Avatar`
- Add `MessageDialog`
- Add `PreferencesWindow`, `PreferencesPage` and `PreferencesGroup`
- Add `ActionRow`, `ComboRow`, `ExpanderRow`, `EntryRow`, `PasswordEntryRow`, and `Preferences Row`
- Add `AboutWindow`
- Add `TabButton`, `TabView`, `TabBar`, `TabOverview`
- Add `Clamp`
- Add `ButtonContent` and a new initializer for the `Button`, supporting icon buttons and buttons with both an icon and a label
- Add type-safe support for system icons (`Icon`) for some widgets, licenses (`License`) for the `AboutWindow`, and edges (`Edge`) for the `NativeWidgetPeer`'s new `padding(_:_:)` function

</details>

### CODE REVIEW

**Comment by JCWasmx86** [comment]
First of all: I'm speechless. That's incredible work!


Just some issues I found:
- Closing the window from `swift run SwiftGui` doesn't end the process
- Memory does not seem to be deallocated (Well, it's in my branch neither :p)
- Style issues, but I can fix that, too.

Except the first one, I'm fine with merging.

File: Sources/SwiftGui/SwiftGui.swift [hunk 1]
```diff
@@ -14,24 +14,188 @@ public class MyApplication: Application {
...
+    }
+  }
+
+  func aboutWindow(win: Window) -> AboutWindow {
+    AboutWindow(app: self, parent: win)
+      .generalData(title: "SwiftGui", icon: .default(icon: .applicationXExecutable), developer: "JCWasmx86", version: "main")
+      .website(url: "https://github.com/JCWasmx86/SwiftGui")
+      .issue(url: "https://github.com/JCWasmx86/SwiftGui/issues")
+      .developers(["JCWasmx86 https://github.com/JCWasmx86", "david-swift https://github.com/david-swift"])
+      .copyright("© 2023 JCWasmx86")
```
Comment (JCWasmx86) [inline Sources/SwiftGui/SwiftGui.swift]: Here you can add yourself too, as you wrote more code than I did ^^ (BTW, should I make you a maintainer? SO you can instantly push to main?)

**Comment by david-swift** [comment]
- Closing the last window should now terminate the process (I called the wrong initializer in child classes of `Window`).
- I do not have any experience with manual memory management, so I do not know how to deallocate the memory.
- What do you mean by "style issues"?

About the copyright: How would you add multiple persons to a copyright, like that: `© 2023 JCWasmx86, david-swift`?

I do not need to be a maintainer, as I would open PRs anyway.

**Comment by JCWasmx86** [comment]
> I do not have any experience with manual memory management, so I do not know how to deallocate the memory.

No worries, that's for sometime else :)

> What do you mean by "style issues"?

I forgot to attach a config file for formatting (swift-format) and swiftlint. But that's irrelevant for this PR. I will fix it up afterwards

> About the copyright: How would you add multiple persons to a copyright, like that: `© 2023 JCWasmx86, david-swift`?

Yes

**Pull request merged**